Sources/JsonHelper.swift

```diff
@@ -0,0 +1,213 @@
+     
+     - author: FelixSFD
+     */
+    public static func encode(object: [String: Any]) -> EncodedDictionary {
+        var dict = object
+        for key in dict.keys {
+            if dict[key] is Date {
+                dict[key] = JsonHelper.encode(object: dict[key] as! Date)
+            }
+            
+            if dict[key] is DictionaryConvertible {
+                dict[key] = JsonHelper.encode(object: dict[key] as! DictionaryConvertible)
+            }
+            
+            if dict[key] is StringRepresentable {
+                dict[key] = JsonHelper.encode(object: dict[key] as! StringRepresentable)
+            }
+            
+            if dict[key] is URL {
+                dict[key] = JsonHelper.encode(object: dict[key] as! URL)
+            }
+        }
+        
+        return dict
+    }
+    
+    /**
+     Encodes the contents of an array recursively.
+     
+     - author: FelixSFD
+     
+     - seealso `encode(object: [String: Any])`
+     */
+    public static func encode(object: [Any]) -> EncodedArray {
+        var array = object
+        for (index, item) in array.enumerated() {
+            if array[index] is Date {
+                array[index] = JsonHelper.encode(object: item as! Date)
+            }
+            
+            if array[index] is DictionaryConvertible {
+                array[index] = JsonHelper.encode(object: item as! DictionaryConvertible)
+            }
+            
+            if array[index] is StringRepresentable {
+                array[index] = JsonHelper.encode(object: item as! StringRepresentable)
+            }
+            
+            if array[index] is URL {
+                array[index] = JsonHelper.encode(object: item as! URL)
+            }
+        }
+        
+        return array
+    }
+    
+    /**
+     Encodes dictionary convertibles like `User`.
+     */
+    public static func encode(object: DictionaryConvertible) -> EncodedDictionary {
+        return JsonHelper.encode(object: object.dictionary)
+    }
+    
+    /**
+     Encodes objects that conform to `StringRepresentable`.
+     */
+    public static func encode(object: StringRepresentable) -> String {
+        return object.rawValue
+    }
+    
+    
+    /**
+     Returns the UNIX timestamp from the `Date`
+     */
+    public static func encode(object: Date) -> Int {
+        return Int(object.timeIntervalSince1970)
+    }
+    
+    /**
+     Returns the absolute string of the `URL`.
+     */
+    public static func encode(object: URL) -> String {
+        return object.absoluteString
+    }
+    
+    // - MARK: Encode to JSON-string
+    
+    /**
+     Encodes a `[String: Any]` to a valid JSON-string.
+     
+     - parameter dictionary: The dictionary to encode
+     
+     - returns: The JSON-string or `nil` if an error occurred
+     
+     - author: FelixSFD
+     */
+    public static func jsonString(from dictionary: [String: Any]) -> String? {
```

Comment:
Nice work!
I think functions like this should `throw` instead of returning an optional.  There are quite a few errors that could occur while parsing JSON, and by throwing the error, we'll know *what* the error was, not just that there was an error.  If we just want the optional result, we can just use `try?`.